src/utils/DateUtils.ts

```diff
@@ -0,0 +1,116 @@
+  static currentYear: number = DateUtils.currentDate.getFullYear();
+
+  static getPreviousYears(numberOfYears: number): string[] {
+    let years: string[] = [];
+
+    for (let year = DateUtils.currentYear - 1; year >= DateUtils.currentYear - numberOfYears; year--) {
+      years.push(year.toString());
+    }
+
+    return years.reverse();
+  }
+
+  static getNextYears(numberOfYears: number): string[] {
+    let years: string[] = [];
+
+    for (let year = DateUtils.currentYear + 1; year <= DateUtils.currentYear + numberOfYears; year++) {
+      years.push(year.toString());
+    }
+
+    return years;
+  }
+
+  static getMonthWeeks(firstDay: Date, startingDay: number): IDay[][] {
+    let weeks: IDay[][] = [];
+    let done = false;
+    let date = moment(firstDay).startOf('week').add(startingDay, 'day');
+    let monthIndex = date.month();
+    let count = 0;
+
+    while (!done) {
+      let days: IDay[] = [];
+      for (let i = 0; i < 7; i++) {
+        days.push({
+          number: date.date(),
+          isCurrentMonth: date.month() === firstDay.getMonth(),
+          isToday: date.isSame(DateUtils.currentDate, 'day'),
+          date: date
+        });
+        date = date.clone();
+        date.add(1, 'day');
+      }
+      weeks.push(days);
+      done = count++ > 2 && monthIndex !== date.month();
+      monthIndex = date.month();
+    }
+
+    return weeks;
+  }
+
+  static getDateWithTimeString(date: Date): string {
+    let dateMoment = moment(date);
+    if (!date || !dateMoment.isValid()) {
+      return '';
+    }
+    return dateMoment.format(LONG_DATE_FORMAT);
+  }
+
+  static getDateFromTimeString(date: string): Date {
+    return moment(date, LONG_DATE_FORMAT, true).toDate();
+  }
+
+  static getSimpleDate(date: Date): string {
+    let dateMoment = moment(date);
+    if (!date || !dateMoment.isValid()) {
+      return '';
+    }
+    return dateMoment.format(SIMPLE_DATE_FORMAT);
+  }
+
+  static getDateFromSimpleDateString(date: string): Date {
+    return moment(date, SIMPLE_DATE_FORMAT, true).toDate();
+  }
+
+  static getRelativeDate(startingDate: Date, dateModifiers: IDateComponents): Date {
+    _.defaults(dateModifiers, {
+      year: 0,
+      month: 0,
+      date: 0,
+      hours: 0,
+      minutes: 0,
+      seconds: 0,
+      milliseconds: 0
+    });
+
+    let year: number = startingDate.getFullYear() + dateModifiers.year;
+    let month: number = startingDate.getMonth() + dateModifiers.month;
+    let date: number = startingDate.getDate() + dateModifiers.date;
+    let hours: number = startingDate.getHours() + dateModifiers.hours;
+    let minutes: number = startingDate.getMinutes() + dateModifiers.minutes;
+    let seconds: number = startingDate.getSeconds() + dateModifiers.seconds;
+    let milliseconds: number = startingDate.getMilliseconds() + dateModifiers.milliseconds;
+
+    return new Date(year, month, date, hours, minutes, seconds, milliseconds);
+  }
```

Comment:
I have been using it (add and subtract) elsewhere to simplify date manipulation. I have no idea why I did not think of using it here. I deleted the method and replaced it by subtract where used.